cmd/livepeer/livepeer.go

```diff
@@ -639,6 +639,20 @@ func main() {
 
 		bcast := core.NewBroadcaster(n)
 
+		var dbOrchPoolCache *discovery.DBOrchestratorPoolCache
+		// When the node is on-chain mode always cache the on-chain orchestrators and poll for updates
+		// Right now we rely on the DBOrchestratorPoolCache constructor to do this. Consider separating the logic
+		// caching/polling from the logic for fetching orchestrators during discovery
+		if *network != "offchain" {
+			var err error
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
There are seven calls to `context.Background()` in this function.  This is wrong.  Please assign
```ctx := context.Background()```
...once at the top of the function, and you can
```
ctx, cancel := context.WithCancel(ctx)
```
...to fork the context as necessary, all seven times.

This covers the `main()` function, but all the others will also need to be modified to take a forked context out of main, rather than creating a new `context.Background()`

```
cmd/livepeer/livepeer.go:		chainID, err := backend.ChainID(context.Background())
cmd/livepeer/livepeer.go:			ctx, cancel := context.WithCancel(context.Background())
cmd/livepeer/livepeer.go:			gasPriceUpdate, err := gpm.Start(context.Background())
cmd/livepeer/livepeer.go:			rs.Start(context.Background())
cmd/livepeer/livepeer.go:		blockWatchCtx, cancel := context.WithCancel(context.Background())
cmd/livepeer/livepeer.go:			ctx, cancel := context.WithCancel(context.Background())
cmd/livepeer/livepeer.go:	msCtx, cancel := context.WithCancel(context.Background())
common/testutil.go:	return peer.NewContext(context.Background(), p)
core/orchestrator.go:	return context.WithTimeout(context.Background(), transcodeLoopTimeout)
core/orchestrator.go:			ctx, cancel := context.WithTimeout(context.Background(), transcodeLoopTimeout)
core/orchestrator.go:	ctx, cancel := context.WithTimeout(context.Background(), RemoteTranscoderTimeout)
discovery/db_discovery.go:	ctx, cancel := context.WithTimeout(context.Background(), getOrchestratorsTimeoutLoop)
discovery/db_discovery.go:	ctx, cancel := context.WithTimeout(context.Background(), getOrchestratorsTimeoutLoop)
discovery/discovery.go:	ctx, cancel := context.WithTimeout(context.Background(), getOrchestratorsTimeoutLoop)
eth/blockwatch/client.go:	ctx, cancel := context.WithTimeout(context.Background(), rc.requestTimeout)
eth/blockwatch/client.go:	ctx, cancel := context.WithTimeout(context.Background(), rc.requestTimeout)
eth/blockwatch/client.go:	ctx, cancel := context.WithTimeout(context.Background(), rc.requestTimeout)
eth/client.go:	chainID, err := eth.ChainID(context.Background())
eth/client.go:	ctx, cancel := context.WithTimeout(context.Background(), c.txTimeout)
eth/client.go:	_, pending, err := c.backend.TransactionByHash(context.Background(), tx.Hash())
eth/client.go:		suggestedGasPrice, err := c.backend.SuggestGasPrice(context.Background())
eth/client.go:	err = c.backend.SendTransaction(context.Background(), newSignedTx)
eth/eventservices/rewardservice.go:	cancelCtx, cancel := context.WithCancel(context.Background())
eth/noncemanager.go:	remoteNonce, err := m.remoteReader.PendingNonceAt(context.Background(), addr)
monitor/census.go:	census.ctx, err = tag.New(context.Background(), tag.Insert(census.kNodeType, nodeType), tag.Insert(census.kNodeID, nodeID))
monitor/census.go:	ctx, err := tag.New(context.Background(), tag.Insert(census.kNodeType, nodeType), tag.Insert(census.kNodeID, nodeID),
server/mediaserver.go:	lpmsCtx, cancel := context.WithCancel(context.Background())
server/mediaserver.go:			err := s.RTMPSegmenter.SegmentRTMPToHLS(context.Background(), rtmpStrm, hlsStrm, segOptions)
server/ot_rpc.go:	ctx := context.Background()
server/rpc.go:	ctx, cancel := context.WithTimeout(context.Background(), GRPCTimeout)
server/webserver.go:			balance, err := b.BalanceAt(context.Background(), s.LivepeerNode.Eth.Account().Address, nil)
server/webserver.go:			blk, err := backend.BlockByNumber(context.Background(), nil)
server/webserver.go:		chainID, err := be.ChainID(context.Background())
```
...We will have to handle these others in a separate PR - no need to muck up this one.